src/plugins/intel_gpu/src/graph/non_zero.cpp

```diff
@@ -0,0 +1,119 @@
+    return layout{cldnn::data_types::i32, cldnn::format::bfyx, tensor{1,1,1,4}};
+}
+
+std::string count_nonzero_inst::to_string(count_nonzero_node const& node) {
+    auto desc = node.get_primitive();
+    auto node_info = node.desc_to_json();
+    auto& input = node.input();
+
+    std::stringstream primitive_description;
+
+    json_composite count_nonzero_info;
+    count_nonzero_info.add("input id", input.id());
+    count_nonzero_info.add("output shape", tensor{1,1,1,4});
+
+    node_info->add("count_nonzero info", count_nonzero_info);
+    node_info->dump(primitive_description);
+
+    return primitive_description.str();
+}
+
+count_nonzero_inst::typed_primitive_inst(network& network, count_nonzero_node const& node) : parent(network, node) {}
+
+static std::vector<int64_t> read_vector(cldnn::memory::ptr mem, cldnn::stream& stream) {
+    switch (mem->get_layout().data_type) {
+        case data_types::i32: {
+            mem_lock<int32_t, mem_lock_type::read> lock{mem, stream};
+            return std::vector<int64_t>(lock.begin(), lock.end());
+        }
+        case data_types::i64: {
+            mem_lock<int64_t, mem_lock_type::read> lock{mem, stream};
+            return std::vector<int64_t>(lock.begin(), lock.end());
+        }
+        default: IE_THROW() << "read_vector: unsupported data type";
+    }
+}
+
+// -----------------------------------------------
+// gather_nonzero
+// -----------------------------------------------
+primitive_type_id gather_nonzero::type_id() {
+    static primitive_type_base<gather_nonzero> instance;
+    return &instance;
+}
+
+layout gather_nonzero_inst::calc_output_layout(gather_nonzero_node const& node) {
+    assert(static_cast<bool>(node.get_primitive()->output_data_type) == false &&
+           "Output data type forcing is not supported for gather_nonzero_node!");
+    auto prim = node.get_primitive();
+    return layout{cldnn::data_types::i32, cldnn::format::bfyx, prim->output_shape};
+}
+
+std::string gather_nonzero_inst::to_string(gather_nonzero_node const& node) {
+    auto desc = node.get_primitive();
+    auto node_info = node.desc_to_json();
+    auto& input = node.input();
+
+    std::stringstream primitive_description;
+
+    json_composite gather_nonzero_info;
+    gather_nonzero_info.add("input id", input.id());
+    gather_nonzero_info.add("output shape", desc->output_shape);
+
+    node_info->add("gather_nonzero info", gather_nonzero_info);
+    node_info->dump(primitive_description);
+
+    return primitive_description.str();
+}
+
+gather_nonzero_inst::typed_primitive_inst(network& network, gather_nonzero_node const& node) : parent(network, node, false) {}
+
+void gather_nonzero_inst::update_shape() {
+    auto& node = const_cast<gather_nonzero_node&>(dynamic_cast<const gather_nonzero_node&>(_node));
+    if (_node.get_dependencies().size() == 2) {
+        auto shape_mem = _network.get_output_memory(_node.get_dependency(1).id());
```

Comment:
I think it should have 2 deps always

Reply:
I removed the if statement @ line 99. Thank you!